docs: restructure documentation (#370)
elastic/apm-agent-ios: docs/reference/edot-ios/configuration.md

```swift
let config = AgentConfigBuilder()
                .withServerUrl(URL(string: "http://localhost:8200"))
                .withSecretToken("<Token>")
                .build()
import ElasticApm
import Foundation

let configuration = AgentConfigBuilder()
  .withExportUrl(URL(string: "https://your-otlp-endpoint")!)
  .withApiKey("your-api-key")
  .useConnectionType(.http)
  .build()

ElasticApmAgent.start(with: configuration)
```

### Export connectivity [export-connectivity]

ElasticApmAgent.start(with:config)
Configure where EDOT iOS exports telemetry and which OTLP transport it uses:

```swift
let configuration = AgentConfigBuilder()
  .withExportUrl(URL(string: "https://collector.example.com:4318")!) // <1>
  .useConnectionType(.http) // <2>
  .build()
```

## Configuration options [configuration-options]
1. The base endpoint that receives OTLP data.
2. The OTLP transport. Use `.http` for OTLP/HTTP or `.grpc` for OTLP/gRPC.

#### `withExportUrl(_:)` [withExportUrl]

| Type | Required |
| --- | --- |
| `URL` | Yes |

You can configure the `AgentConfigBuilder` with the following functions.
Sets the OTLP endpoint provided by an Elastic Agent or EDOT Collector gateway.

> ### `withServerUrl` [withServerUrl] **Deprecated**

Reviewer: This was removed from the new docs, but if it’s still available to users,  I think we shouldn’t be silently deleting it (for the sake of users on older versions and those still using this config).

Instead, we can use a section-level `applies_to` to add a deprecated label, and we could also even say when it would be removed.

So something like (I’ve just added random version, not sure when this was deprecated or if/when it will be removed):

````
### `withServerUrl` [withServerUrl]

```{applies_to}
product:
  edot_ios: deprecated 1.4.0
```
````
Rendered as “Deprecated since 1.4”:
<img width="435" height="357" alt="Screenshot 2026-07-31 at 11 06 18" src="https://github.com/user-attachments/assets/c91d5911-0e6e-4128-a47a-b92fc91339c2" />

or

````
### `withServerUrl` [withServerUrl]

```{applies_to}
product:
  edot_ios: deprecated 1.2-2.0, removed 2.1.0
```
````
Rendered as "Deprecated from 1.2-2.0”, but on hover shows “Planned for removal” until 2.1.0 is released:
<img width="440" height="390" alt="Screenshot 2026-07-31 at 11 01 44" src="https://github.com/user-attachments/assets/2b70c539-74fb-45b2-8d98-b2567667a864" />

or 

````
### `withServerUrl` [withServerUrl]

```{applies_to}
product:
  edot_ios: deprecated 1.2-2.1, removed 2.2.0
```
````

Rendered as “Deprecated since 1.2”, on hover shows “Planned for removal” until 2.1.0 is released. After 2.1.0 is released, it would show "Deprecated from 1.2-2.1” and “Planned for removal” on hover:
<img width="437" height="395" alt="Screenshot 2026-07-31 at 11 02 10" src="https://github.com/user-attachments/assets/98ce6a19-66c8-4fd6-a13b-ec6d98595125" />

Author: Thanks! I've re-added `withServerUrl` with a section-level `applies_to` deprecated label. I checked the code history and the deprecation shipped in 2.0.0, so I used `deprecated 2.0.0` (there's no planned removal version in the code yet). I also applied the same treatment to `withAppMetricInstrumentation`, which is still available as a deprecated no-op since 2.0.1.

If `useOpAMP` is enabled, set the URL to your OpAMP endpoint, such as `http://localhost:4320/v1/opamp`. For example: 
Sets an APM agent key and sends it in the `Authorization` header using the `ApiKey` scheme:

```swift
let config = AgentConfigBuilder()
                .withServerUrl(URL(string: "http://localhost:8200")!)
                .withManagementUrl(URL(string:"http://localhost:4320/v1/opamp")!)
                .useOpAMP()
                .build()
let configuration = AgentConfigBuilder()
  .withExportUrl(URL(string: "https://your-otlp-endpoint")!)
  .withApiKey("your-api-key")
  .build()
```

Create an [APM agent key for EDOT SDKs](docs-content://solutions/observability/apm/opentelemetry/create-apm-agent-key-for-edot-sdks.md) to use least-privilege credentials.

#### `withSecretToken(_:)` [secretToken]

ElasticApmAgent.start(with:config)
| Type | Default |
| --- | --- |
| `String` | No authentication |

Sets an APM secret token and sends it in the `Authorization` header using the `Bearer` scheme:

```swift
let configuration = AgentConfigBuilder()
  .withExportUrl(URL(string: "https://your-otlp-endpoint")!)
  .withSecretToken("your-secret-token")
  .build()
```
:::

> ### `withRemoteManagement` [withRemoteManagement]

Reviewer: Claude highlighted this:

> `withRemoteManagement` is undocumented but still a live, non-deprecated API `Sources/apm-agent-ios/AgentConfigBuilder.swift:62` still exposes `withRemoteManagement(_:)` with no `@available(deprecated)` marker, but the new docs never mention it.

So shouldn't this be in the new docs, too? Or could it also use a deprecated label (a section-level `applies_to` similar to [my other comment](https://github.com/elastic/apm-agent-ios/pull/370/changes#r3688929127))?

Author: Thanks, good catch! `withRemoteManagement` is still a live, non-deprecated API, so I've documented it under the Remote management connectivity section.

| --- | --- |
| Method call | Disabled |

Sets the API Token for connecting to an authenticated APM Server. If using the env-var, the whole header map must be defined per [OpenTelemetry Protocol Exporter Config](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) (for example: `OTEL_EXPORTER_OTLP_HEADERS="Authorization=ApiKey <key>"`)
Enables OpAMP-based central configuration. When enabled, EDOT iOS treats `withManagementUrl(_:)` as an OpAMP endpoint.

This setting is mutually exclusive with `withSecretToken`
Refer to [Central configuration](#central-configuration) for the complete setup.

### `useConnectionType` [useConnectionType]
### Session behavior [session-behavior]

* **Type:** `AgentConnectionType`
* **Default:** `.grpc`
EDOT iOS makes a span sampling decision when a new app session begins.

Selects the transport used to export OTLP data to the collector. `.grpc` uses the gRPC OTLP exporter (default). `.http` uses the OTLP/HTTP exporters and will send traces, metrics, and logs to the HTTP endpoints (for example `/v1/traces`, `/v1/metrics`, `/v1/logs`). 
#### `withSessionSampleRate(_:)` [withSessionSampleRate]

> ### TLS connections [tls-connections]

Reviewer: This section was removed after the restructuring deliberately?

Author: Yes, it doesn't provide a configuration option, and it's basically stating that the agent can talk to "https" urls, which I think is not worth mentioning. But I'm open to reverting it in case it might be necessary for cases I haven't taken into account.

Reviewer: Ok, let’s keep your changes.

> Sets the probability that spans from a new session are sampled. A value of `1.0` samples every session; `0.0` samples none. Values outside this range are clamped to the closest valid value.

Reviewer: ```suggestion
Sets the probability that spans from a new session are sampled. A value of `1.0` samples every session, `0.0` samples none. Values outside this range are clamped to the closest valid value.
```

### Deployment environment [deployment-environment]

EDOT iOS sets `deployment.environment.name` to `default`. Override it with `OTEL_RESOURCE_ATTRIBUTES`:

```text
deployment.environment.name=staging
```

## Dynamic configuration [dynamic-configuration]

Dynamic settings can change after EDOT iOS starts. The SDK retrieves them through [Central configuration](#central-configuration).

### Recording [recording]

Controls whether EDOT iOS processes and exports log records. Central configuration polling remains active when recording is disabled.

| Default | Type | Dynamic |
| --- | --- | --- |
| `true` | Boolean | Yes |

### Session sample rate [session-sample-rate]

Controls the probability that spans from a new session are sampled. The value can range from `0.0` to `1.0`.

The setting is evaluated when a new session begins. This keeps related spans together instead of making a separate decision for every span.

| Default | Type | Dynamic |
| --- | --- | --- |
| `1.0` | Double | Yes |

## Central configuration [central-configuration]

> EDOT iOS receives central configuration from an EDOT Collector through OpAMP.
> 
> ```{applies_to}
> product:
>   edot_ios: preview 1.4.0
> ```

Reviewer: Having the intro sentence before the `applies_to` tag breaks the section-level `applies_to` rendering, so this should be:

````suggestion
## Central configuration [central-configuration]

```{applies_to}
product:
 edot_ios: preview 1.4.0
```

EDOT iOS receives central configuration from an EDOT Collector through OpAMP.
````

Author: Thanks, I've applied the change — the `applies_to` block now sits directly under the heading.

Refer to [Central configuration](#central-configuration) for the complete setup.

#### `withRemoteManagement(_:)` [withRemoteManagement]

| Type | Default |
| --- | --- |
| `Bool` | `true` |

> Controls whether EDOT iOS contacts central configuration for runtime configuration updates. Set it to `false` to disable remote management entirely.

Reviewer: ```suggestion
Controls whether EDOT iOS contacts central configuration for runtime configuration updates. Set it to `false` to turn off remote management entirely.
```

Author: Thanks, I've applied the change.
